homeassistant/components/cover/deconz.py

```diff
@@ -29,30 +30,35 @@
 
     Covers are based on same device class as lights in deCONZ.
     """
+    gateway = hass.data[DECONZ_DOMAIN]
+
     @callback
     def async_add_cover(lights):
         """Add cover from deCONZ."""
         entities = []
         for light in lights:
             if light.type in COVER_TYPES:
                 if light.modelid in ZIGBEE_SPEC:
-                    entities.append(DeconzCoverZigbeeSpec(light))
+                    entities.append(DeconzCoverZigbeeSpec(light, gateway))
                 else:
-                    entities.append(DeconzCover(light))
+                    entities.append(DeconzCover(light, gateway))
         async_add_entities(entities, True)
 
-    hass.data[DATA_DECONZ].listeners.append(
+    gateway.listeners.append(
```

Comment:
I like you moved these kind of stuff into the gateway

Reply:
Shamelessly stolen from Balloobs Hue implementation ;)